sycl/doc/SYCLGraphUsageGuide.md

```diff
@@ -0,0 +1,493 @@
+bundle with `handler::set_args()` and having its node arguments updated.
+
+```c++
+...
+
+using namespace sycl;
+namespace sycl_ext = sycl::ext::oneapi::experimental;
+
+queue myQueue;
+auto myContext = myQueue.get_context();
+auto myDevice = myQueue.get_device();
+
+// USM allocations for kernel input/output
+const size_t n = 1024;
+int *ptrX = malloc_shared<int>(n, myQueue);
+int *ptrY = malloc_device<int>(n, myQueue);
+
+int *ptrZ = malloc_shared<int>(n, myQueue);
+int *ptrQ = malloc_device<int>(n, myQueue);
+
+// Kernel loaded from kernel bundle
+const std::vector<kernel_id> builtinKernelIds =
+      myDevice.get_info<info::device::built_in_kernel_ids>();
+kernel_bundle<bundle_state::executable> myBundle =
+      get_kernel_bundle(myContext, { myDevice }, builtinKernelIds);
+kernel builtinKernel = myBundle.get_kernel(builtinKernelIds[0]);
+
+// Graph containing a two kernels node
+sycl_ext::command_graph myGraph(myContext, myDevice);
+
+int myScalar = 42;
+// Create graph dynamic parameters
+dynamic_parameter dynParamInput(myGraph, ptrX);
+dynamic_parameter dynParamScalar(myGraph, myScalar);
+
+// First node uses ptrX as an input & output parameter, with operand
+// mySclar as another argument.
+node nodeA = myGraph.add([&](handler& cgh) {
+    cgh.set_args(dynParamInput, ptrY, dynParamScalar);
+    cgh.parallel_for(range {n}, builtinKernel);
+});
+
+// Create an executable graph with the updatable property.
+auto execGraph = myGraph.finalize({sycl_ext::property::graph::updatable});
+
+// Execute graph, then update without needing to wait for it to complete
+myQueue.ext_oneapi_graph(execGraph);
+
+// Change ptrX argument to node A to ptrZ
+dynParamInput.update(ptrZ);
+
+// Change myScalar argument to node A to newScalar
+int newScalar = 12;
+dynParamScalar.update(newScalar);
+
+// Update nodeA in the executable graph with the new parameters
+execGraph.update(nodeA);
+// Execute graph again
+myQueue.ext_oneapi_graph(execGraph);
+myQueue.wait();
+
+sycl::free(ptrX, myQueue);
+sycl::free(ptrY, myQueue);
+sycl::free(ptrZ, myQueue);
+sycl::free(ptrQ, myQueue);
+
+```
+
+Example snippet showing how to use accessors with `dynamic_parameter` update:
+
+```c++
+sycl::buffer bufferA{...};
+sycl::buffer bufferB{...};
+
+// Create graph dynamic parameter using a placeholder accessor, since the
+// sycl::handler is not available here outside of the command-group scope.
+dynamic_parameter dynParamAccessor(myGraph, bufferA.get_access());
+
+node nodeA = myGraph.add([&](handler& cgh) {
+    // Require the accessor contained in the dynamic paramter
+    cgh.require(dynParamAccessor);
+    // Set the arg on the kernel using the dynamic parameter directly
+    cgh.set_args(dynParamAccessor);
+    cgh.parallel_for(range {n}, builtinKernel);
+});
+
+...
+// Update the dynamic parameter with a placeholder accessor from bufferB instead
+dynParamAccessor.update(bufferB.get_access());
+```
+
+### Whole Graph Update
+
```

Comment:
If we can link to the whole graph update section of the spec here it might help the reader.